src/events/api/views.py

```diff
@@ -7,84 +7,92 @@
 
 from django.conf import settings
 from django.db.models import Count
-from django.shortcuts import get_object_or_404
+from django.http import Http404
 
 from core.authentication import TokenAuthentication
 from events.models import (
     CustomEvent, Location, ProposedTalkEvent,
     ProposedTutorialEvent, SponsoredEvent, Time, KeynoteEvent
 )
-from proposals.models import TalkProposal, TutorialProposal
 
 from . import serializers
 
 
-class TalkDetailAPIView(RetrieveAPIView):
-    authentication_classes = [TokenAuthentication]
-    permission_classes = [IsAuthenticated]
+class TalkListAPIView(ListAPIView):
+    queryset = ProposedTalkEvent.objects.all()
+    serializer_class = serializers.TalkListSerializer
 
-    def get_serializer_class(self):
-        is_sponsored = self.request.GET.get('isSponsored')
-        if is_sponsored:
-            return serializers.SponsoredEventDetailSerializer
-        return serializers.TalkDetailSerializer
 
-    def get_queryset(self):
-        is_sponsored = self.request.GET.get('isSponsored')
-        if is_sponsored:
-            return SponsoredEvent.objects.all()
-        return ProposedTalkEvent.objects.all()
+class SponsoredEventListAPIView(ListAPIView):
+    queryset = SponsoredEvent.objects.all()
+    serializer_class = serializers.SponsoredEventListSerializer
 
-    def get_object(self):
-        is_sponsored = self.request.GET.get('isSponsored')
-        pk = self.kwargs["pk"]
-        queryset = self.get_queryset()
-        if is_sponsored:
-            return get_object_or_404(queryset, id=pk)
-        return get_object_or_404(queryset, proposal_id=pk)
 
+class TutorialListAPIView(ListAPIView):
+    queryset = ProposedTutorialEvent.objects.all()
+    serializer_class = serializers.TutorialListSerializer
 
-class TalkListAPIView(APIView):
+
+class SpeechListAPIView(APIView):
     authentication_classes = [TokenAuthentication]
     permission_classes = [IsAuthenticated]
 
-    speech_querysets = [
-        TalkProposal.objects.filter_accepted(),
-        SponsoredEvent.objects.all()
-    ]
+    def get(self, request, *args, **kwargs):
+        event_type_string = request.GET.get("event_types")
+        event_types = event_type_string.split(',') if event_type_string else []
+        data = []
+        for event_type, api_view in [
+            ('talk', TalkListAPIView),
+            ('sponsored', SponsoredEventListAPIView),
+            ('tutorial', TutorialListAPIView),
+        ]:
+            if event_type not in event_types or not event_types:
+                continue
+            view = api_view.as_view()
+            event_data = view(request._request, *args, **kwargs).data
```

Comment:
I suggest use dict to get the API view model, and pass the request directly.
```suggestion
        api_view_dict = {
            'talk': TalkListAPIView,
            'sponsored': SponsoredEventListAPIView,
            'tutorial': TutorialListAPIView,
        }

        for event_type in event_types:
            try:
                api_view = api_view_dict[event_type]
            except KeyError:
                continue
            view = api_view.as_view()
            event_data = view(request, *args, **kwargs).data
```